cdap-api/src/main/java/co/cask/cdap/api/data/stream/StreamBatchReadable.java

```diff
@@ -223,9 +223,22 @@ public StreamBatchReadable(String streamName, long startTime, long endTime) {
    */
   public StreamBatchReadable(String streamName, long startTime,
                              long endTime, Class<? extends StreamEventDecoder> decoderType) {
+    this(streamName, startTime, endTime, decoderType.getName());
+  }
+
+  /**
+   * Constructs an instance with the given properties.
+   *
+   * @param streamName Name of the stream
+   * @param startTime Start timestamp in milliseconds (inclusive) of stream events provided to the job
+   * @param endTime End timestamp in milliseconds (exclusive) of stream events provided to the job
+   * @param decoderTypeClassname The full qualified name of the class for decoding {@link StreamEvent}
+   */
+  public StreamBatchReadable(String streamName, long startTime,
+                             long endTime, String decoderTypeClassname) {
```

Comment:
Shouldn't need this method. See my comment below about not to use this method to set a CDAP internal class name.